src/goose.rs

```diff
@@ -911,7 +912,9 @@ pub struct GooseUser {
     /// Normal tasks are optionally throttled, test_start and test_stop tasks are not.
     pub is_throttled: bool,
     /// Channel to parent.
-    pub parent: Option<mpsc::UnboundedSender<GooseRawRequest>>,
+    pub parent_request_stats: Option<mpsc::UnboundedSender<GooseRawRequest>>,
+    /// Channel to parent.
+    pub parent_task_stats: Option<mpsc::UnboundedSender<GooseRawTask>>,
```

Comment:
I thought about it and we definitely want the enum solution:

enum GooseMetric {
  Request(GooseRawRequest),
  Task(GooseRawTask),
  TaskSet(GooseRawTaskSet)
}
```

etc.

The storage is pretty similar, so an enum is the perfect way to always can just do:

```
parent.send(&GooseMetric::Request(req));
```

etc.

I feel it is worth it to simplify it and make it even simpler to add other statistics in the future.

Maybe even Boxed user defined ones.

Reply:
We now use Enums both for User threads pushing metrics to the Parent thread, as well as for Worker processes pushing metrics to the Manager process.